### DISCUSSION TITLE
ModuleFederationPlugin dosen't work with shared react

### DISCUSSION THREAD

**Discussion by MaxusAdmin** [webpack/webpack#13516] [General]
# Bug report

**What is the current behavior?**

i created project for demostation my problem -> [module_federation_test](https://github.com/MaxusAdmin/module_federation_test/tree/master/host_app)

I have two apps.
1. **host_app** is app which render remote app on different port. _it works on localhost:8080_
2. **remote_app** is app which rende only - hello world, good luck. _it  works on localhost:8081_

And my projec didn't work.

I had the problems:
```diff
- 1. Uncaught Error: Shared module is not available for eager consumption: webpack/sharing/consume/default/react/react?024c
- 2. Uncaught Error: Shared module is not available for eager consumption: webpack/sharing/consume/default/react/react
- 3. Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons
```
I tried every possible option, but i didn't solve my problem.

**What is the expected behavior?**

#### config host_app (only moduel federation)
webpack version: 5.37.0
react version: 17.0.2
react-dom version: 17.0.2
```js script 
const AutomaticVendorFederation = require("@module-federation/automatic-vendor-federation");
const exclude = ["babel", "webpack"];
const ignoreVersion = ['react', 'react-dom'];
const filename = ''static/js/'';

{
      name: "host",
      filename: `${filename}remoteEntry.js`,
      remotes: {
        remote_app: "remote_app@http://localhost:8081/static/js/remoteEntry.js",
      },
      shared: [
        {
          ...AutomaticVendorFederation({
            exclude,
            ignoreVersion,
            packageJson,
            shareFrom: ["dependencies", "peerDependencies"],
            ignorePatchVersion: true,
          }),
          // react: {
          //   singleton: true,
          //   eager: true,
          //   requiredVersion: deps.react
          // },
          // "react-dom": {
          //   singleton: true,
          //   eager: true,
          //   requiredVersion: deps["react-dom"]
          // }
        }
      ]
    }
```

#### config remote_app (only moduel federation)
webpack version: 5.37.0
react version: 17.0.2
react-dom version: 17.0.2

```js script
const AutomaticVendorFederation = require("@module-federation/automatic-vendor-federation");
const exclude = ["babel", "webpack"];
const ignoreVersion = ['react', 'react-dom'];
const filename = ''static/js/'';
cons name = "remote_app"";
const exposes = {};
{
      name: name,
      library: { type: "var", name: name },
      filename: `${filename}remoteEntry.js`,
      exposes: {
        "./RootApp": "./src/apps/RemoteApp",
        ...exposes
      },
      shared: [
        {
          ...AutomaticVendorFederation({
            exclude,
            ignoreVersion,
            packageJson,
            shareFrom: ["dependencies", "peerDependencies"],
            ignorePatchVersion: true,
          }),
          // react: {
          //   singleton: true,
          //   eager: true,
          //   requiredVersion: deps.react
          // },
          // "react-dom": {
          //   singleton: true,
          //   eager: true,
          //   requiredVersion: deps["react-dom"]
          // }
        }
      ]
    }
```
I need help. Maybe i don't  understand philosophy or other things which are important for solve my problem.
I feel very close to solving my problem, but now not yet ((((

**Comment by alexander-akait** [comment]
Please ask about this in https://github.com/module-federation/automatic-vendor-federation

**Reply by alexander-akait** [reply]
> Shared module is not available for eager consumption

It means you can't use this module for eager consumption, it should not be shared in your case

**Reply by MaxusAdmin** [reply]
If i remove react from shred option i will get error: Invalid hook call

I tried a lots of option, but it didn't work.

I made test project on GitHub (link above in the problem description) when anyone can test their theory of the truthful version code.

p.s. 
I used automatic-vendor-federation because i had a lots of problem with shared library

**Reply by MaxusAdmin** [reply]
![Screenshot_2021-06-04-15-33-18-691_com.android.chrome.jpg](https://user-images.githubusercontent.com/22521593/120801795-4f159b80-c54a-11eb-9a5d-0f769150a4a7.jpg)

**Comment by XiaofengXie16** [comment]
This bug is already resolved :https://github.com/webpack/webpack/issues/13517#issuecomment-855188398

 @alexander-akait I am sorry for the inconvenience  , I accidentally re-opened that issue after leaving that extra comment and misled you.
